src/python/zquantum/core/measurement.py

```diff
@@ -392,3 +393,115 @@ def convert_bitstring_to_int(bitstring: Iterable[int]) -> int:
             significant (little endian).
     """
     return int("".join(str(bit) for bit in bitstring[::-1]), 2)
+
+class Measurements:
+    """ A class representing measurements from a quantum circuit. """
+
+    def __init__(self, filename=None):
+        self.counts = {}
+        self.measurements = []
+        self.num_measurements = 0
+
+        if filename is not None:
+            with open(filename, "r") as f:
+                data = json.load(f)
+                self.add_measurements(data["measurements"])
+
+
+    def save(self, filename):
+        """ Serialize the Measurements object into a file in JSON format.
+        
+        Args:
+            filename (string): filename to save the data to 
+        """
+        data = { "schema":         SCHEMA_VERSION+"-measurements",
+                 "counts":         self.get_counts(),
+                 "measurements":   self.get_measurements()}
+        with open(filename, "w") as f:
+            f.write(json.dumps(data, indent=2))
+
+
+    def get_measurements(self):
+        """ Get the measurements as a list of tuples
+
+        Returns:
+            A list of tuples where the value at a given index is the measured state of the qubt at the given index 
+        """
+        return self.measurements
+
+
+    def get_counts(self):
+        """ Get the measurements as a list of tuples
+
+        Returns:
+            A list of tuples where the value at a given index is the measured state of the qubt at the given index 
+        """
+        return self.counts
+
+
+    def get_num_measurements(self):
+        """ Get the number of measurements
+
+        Returns:
+            integer representing the number of measurements
+        """
+        return self.num_measurements
+
+
+    def add_measurements(self, measurements):
+        """ Add a list of measurement tuples 
+
+        Args:
+            measurements (list of tuples): A list of the measurements to add
+        """
+        for measurement in measurements:
+            self.measurements.append(tuple(measurement))
+
+            # Bitstrings are in reversed-index order compared to measurement tuples
```

Comment:
Why?
I don't think we should have such inconsistencies in the core library.

Reply:
Addressed in refactor here:
231a59a